knime_extension/src/nodes/spatialstatistics.py

```diff
@@ -59,90 +60,125 @@ class spatialWeights:
         """,
         "Queen",
         enum=[
             "Queen",
             "Rook",
-            "Inverse Distance",
             "Binary Distance Band",
-            "K nearest",
+            "Inverse Distance",
             "Lattice",
+            "K nearest",
             "Kernel",
-            "Your own",
+            "Get spatial weights matrix from file",
         ],
     )
     order = knext.IntParameter(
-        "Order",
+        "Order for Queen or Rook",
         """The order of the weight matrix is 1 by default. Users can change the order of the weights, higher order 
         weights will treat further units as neighbors.""",
         1,
     )
 
-    Nearest_k = knext.IntParameter(
-        "Nearest k",
-        "K-nearest are often used for point data. k is the number of the nearest neighbor.",
-        4,
-    )
-
     Threshold = knext.IntParameter(
-        "Threshold",
-        """Distance band weights are often used for point data. The weights within the threshold are 1 and otherwise 0. 
-        Inverse distance weights are often used for point data. The weights within the threshold are distance^-power, 
-        and otherwise 0. The distance is Euclidean distance.""",
+        "Threshold for Inverse Distance or Binary Distance Band",
+        """The distance threshold for constructing binary distance band and inverse distance weights. Defaults to 1""",
         1,
     )
+
     Power = knext.IntParameter(
-        "Power",
-        """Distance band weights are often used for point data. The weights within the threshold are 1 and otherwise 0. 
-        Inverse distance weights are often used for point data. The weights within the threshold are distance^-power, 
-        and otherwise 0. The distance is Euclidean distance.""",
+        "Power for Inverse Distance",
+        """The power for constructing inverse distance weights. Defaults to 1.""",
         1,
     )
+
     Rows = knext.IntParameter(
-        "Rows", "Please choose your rows and colunns of your lattice.", 5
+        "Rows for Lattice",
+        "The number of rows for constructing a lattice spatial weights matrix. Defaults to 5.",
+        5,
     )
     Columns = knext.IntParameter(
-        "Columns", "Please choose your rows and columns of your lattice.", 5
+        "Columns for Lattice",
+        "The number of columns for constructing a lattice spatial weights matrix. Defaults to 5.",
+        5,
     )
-    Your_own_matrix_local_path = knext.StringParameter(
-        "Your own matrix local path",
-        """Please enter the path of the spatial weights matrix in CSV format in the following options. 
-        The weights matrix must be in matrix format and in the order of the samples.""",
-        "",
+
+    # k = knext.IntParameter(
+    #     "K for K nearest or Kernel",
+    #     "k is the number of the nearest neighbor.",
+    #     4,
+    # )
+
+    Nearest_k = knext.IntParameter(
```

Comment:
Load parameter backward compatible. For an example see here: https://github.com/spatial-data-lab/knime-geospatial-extension/blob/main/knime_extension/src/nodes/spatialtool.py#L601